python_modules/dagster/dagster/_core/storage/captured_log_manager.py

```diff
@@ -258,3 +261,104 @@ def unsubscribe(self, subscription: CapturedLogSubscription) -> None:
     def build_log_key_for_run(self, run_id: str, step_key: str) -> Sequence[str]:
         """Legacy adapter to translate run_id/key to captured log manager-based log_key."""
         return [run_id, "compute_logs", step_key]
+
+    def get_log_keys_for_log_key_prefix(self, log_key_prefix: Sequence[str]) -> Sequence[str]:
+        """Returns the logs keys for a given log key prefix.This is determined by looking at the
+        directory defined by the log key prefix and creating a log_key for each file in the directory.
+        """
+        raise NotImplementedError("Must implement get_log_keys_for_log_key_prefix")
+
+    def _read_json_lines(self, start_line, num_lines, log_lines):
+        records = []
+        if start_line >= len(log_lines):
+            return records, False, 0
+
+        remaining_log_lines = log_lines[start_line:]
+        total_lines_processed = 0
+
+        for line in remaining_log_lines:
+            if len(records) >= num_lines:
+                # got the requested number of lines, end early
+                break
+            total_lines_processed += 1
+            if not line:
+                continue
+            try:
+                records.append(_seven.json.loads(line))
+            except json.JSONDecodeError:
+                continue
+
+        has_more = len(remaining_log_lines) > total_lines_processed
+
+        return records, has_more, total_lines_processed
+
+    def _get_log_lines_for_log_key(self, log_key):
+        log_data = self.get_log_data(log_key)
+        raw_logs = log_data.stderr.decode("utf-8") if log_data.stderr else ""
+        log_lines = raw_logs.split("\n")
+
+        return log_lines
+
+    def read_json_log_lines_for_log_key_prefix(
+        self, log_key_prefix: Sequence[str], cursor: Optional[str], num_lines: int = 100
```

Comment:
I chose 100 at random for the number of lines to fetch by default. what's an actually good default to set? I don't really have anything to go off of. 
For context, most backfill ticks are in the 10s of logs, not 100s

Reply:
well theres this number for one reference 
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster-graphql/dagster_graphql/implementation/execution/__init__.py#L220

so probably can do thousand(s)

env var overridable const is a good way to go